src/transform.js

```diff
@@ -0,0 +1,165 @@
+'use strict';
+const b = require('@babel/core');
+const fs = require('fs');
+
+// addSetupNode adds the setup() call to the beginning of the script.
+function addSetupNode() {
+  const setupNode = b.template.statement(`
+    setup({single_test: true, explicit_done: false});
+  `)();
+  return {
+    visitor: {
+      Program(path) {
+        path.node.body.unshift(setupNode);
+      },
+    },
+  };
+}
+
+const transformBoolMap = {
+  'shouldBeTrue': 'assert_true',
+  'shouldBeFalse': 'assert_false',
+
+};
+
+function transformShouldBeBool() {
+  return {
+    visitor: {
+      CallExpression(path) {
+        if (transformBoolMap.hasOwnProperty(path.node.callee.name)) {
+          path.node.callee.name = transformBoolMap[path.node.callee.name];
+          const newArgument =
+            b.template.expression(path.node.arguments[0].value)();
+          path.node.arguments[0] = newArgument;
+        }
+      },
+    },
+  };
+}
+
+const transformValueMap = {
+  'shouldBeNaN': {
+    name: 'assert_equals',
+    value: b.types.identifier('NaN'),
+  },
+
+  'shouldBeNull': {
+    name: 'assert_equals',
+    value: b.types.nullLiteral(),
+  },
+  'shouldBeNonNull': {
+    name: 'assert_not_equals',
+    value: b.types.nullLiteral(),
+  },
+
+  'shouldBeUndefined': {
+    name: 'assert_equals',
+    value: b.types.identifier('undefined'),
+  },
+  'shouldBeDefined': {
+    name: 'assert_not_equals',
+    value: b.types.identifier('undefined'),
+  },
+
+  'shouldBeZero': {
+    name: 'assert_equals',
+    value: b.types.numericLiteral(0),
+  },
+  'shouldBeNonZero': {
+    name: 'assert_not_equals',
+    value: b.types.numericLiteral(0),
+  },
+
+  'shouldBeEmptyString': {
+    name: 'assert_equals',
+    value: b.types.stringLiteral(''),
+  },
+};
+
+function transformShouldBeValue() {
+  return {
+    visitor: {
+      CallExpression(path) {
+        if (transformValueMap.hasOwnProperty(path.node.callee.name)) {
+          const newArgument1 = transformValueMap[path.node.callee.name].value;
+
+          path.node.callee.name = transformValueMap[path.node.callee.name].name;
+
+          const newArgument0 =
+            b.template.expression(path.node.arguments[0].value)();
+          path.node.arguments = [newArgument0, newArgument1];
+        }
+      },
+    },
+  };
+}
+
+const transformComparatorMap = {
+  'shouldBe': 'assert_equals',
+  'shouldNotBe': 'assert_not_equals',
+  'shouldBeGreaterThan': 'assert_greater_than',
```

Comment:
do `less_than` and `less_than_equal exist`?

Reply:
The new API, testharness.js, has `assert_less_than` and `assert_less_than_equal`, but the old API doesn't have an equivalent for some reason.